Modernize build to use pyproject.toml and scikit-build-core

jbcoe opened the pull request.

jbcoe committed:
Modernize build to use pyproject.toml and scikit-build-core

jbcoe committed:
Fixes for Linux build with uv

jbcoe committed:
Remove python version requirement

jbcoe committed:
Use nle 1.2.0 consistently

jbcoe committed:
Fix python version for compatibility with nle 1.2.0

jbcoe committed:
Install shared library and install nethack shared-library

jbcoe committed:
Tests run, most pass

jbcoe committed:
Tests run, most pass

jbcoe committed:
Use FetchContent and target properties in CMake

jbcoe committed:
Use uv to run tests

jbcoe committed:
Sort includes

jbcoe committed:
Format python

jbcoe committed:
Fix python linting issues

jbcoe committed:
Fix python linting issues

jbcoe committed:
Fix python linting issues

jbcoe committed:
Revert gym to gymnasium

jbcoe committed:
Fix compiler warning

jbcoe committed:
Use fork of NLE 0.8.0 to fix assorted build errors

jbcoe committed:
Set RPATH properly

ngoodger commented:
Hi @jbcoe , thanks for putting this together.  I'm not totally sure how to validate the changes, and get an error (not to say I didn't before!).  Does the README need some updates as well.
I tried running 
```
uv sync
error: No `project` table found in: `/home/ngoodger/Development/nle-language-wrapper/pyproject.toml`
```

jbcoe commented:
There's a fair bit of work still to do here (hence the draft status), I thought I'd open the PR for visibility.

 I was talking to the maintainers of NetHack LE (https://github.com/NetHack-LE/nle) and wondering if this project (nethack language wrapper) is still active?

We'd be keen to update the nethack langauge wrapper to work with latest version of Nethack LE.

jbcoe committed:
Use fork of nethack 0.8.0
Current issues with build

      CMake Error at /usr/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
        Could NOT find Python (missing: Interpreter) (found suitable version
        3.10.18, minimum required is 3.8)

            Reason given by package:
                Interpreter: Cannot run the interpreter /usr/local/google/home/jbcoe/.cache/uv/builds-v0/.tmp3PzyhH/bin/python3

      Call Stack (most recent call first):

        /usr/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
        /usr/share/cmake-3.31/Modules/FindPython/Support.cmake:3997 (find_package_handle_standard_args)
        /usr/share/cmake-3.31/Modules/FindPython.cmake:631 (include)
        build/temp.linux-x86_64-cpython-310/_deps/pybind11-src/tools/pybind11NewTools.cmake:54 (find_package)
        build/temp.linux-x86_64-cpython-310/_deps/pybind11-src/tools/pybind11Common.cmake:204 (include)
        build/temp.linux-x86_64-cpython-310/_deps/pybind11-src/CMakeLists.txt:296 (include)
